Lecture6/Task1/HttpClientsImplementation/ApacheHttpClient.java

```diff
@@ -0,0 +1,62 @@
+package HttpClientsImplementation;
+
+import java.io.IOException;
+
+import HttpClientsImplementation.Utils.RequestDataPreparatory;
+import org.apache.http.HttpResponse;
+import org.apache.http.auth.AuthenticationException;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.impl.auth.BasicScheme;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+public class ApacheHttpClient {
+
+    private static final String ERROR_MESSAGE_IO_EXCEPTION = "Something went wrong (ApacheHttpClient => IOException)";
+    private static final String ERROR_MESSAGE_AUTH_EXCEPTION = "Something went wrong (ApacheHttpClient => AuthenticationException)";
+
+    private ApacheHttpClient() {
+    }
+
+    public static HttpResponse requestGet(RequestDataPreparatory requestData) {
```

Comment:
your low level http client class (that is common for all requests) has knowledge that get request can be done without auth and put doesn't. This is not very good since you design your common class to be used with this particular knowledge. Later, it will be hard to maintain such a class. Instead, add auth request to http clients and make test to auth user if needed. This will make the code more stable and flexible